src/bitwarden_workflow_linter/rules/run_actionlint.py

```diff
@@ -0,0 +1,108 @@
+"""A Rule to run actionlint on workflows."""
+
+from typing import Optional, Tuple
+import subprocess
+import platform
+import urllib.request
+import os
+
+from ..rule import Rule
+from ..models.workflow import Workflow
+from ..utils import LintLevels, Settings
+
+
+def install_actionlint(platform_system: str) -> Tuple[bool, str]:
+    """If actionlint is not installed, detects OS platform
+    and installs actionlint"""
+
+    error = f"An error occurred when installing Actionlint on {platform_system}"
+
+    if platform_system.startswith("Linux"):
+        return install_actionlint_source(error)
+    elif platform_system == "Darwin":
+        try:
+            subprocess.run(["brew", "install", "actionlint"], check=True)
+            return True, ""
+        except (FileNotFoundError, subprocess.CalledProcessError):
+            return False, f"{error} : check Brew installation"
+    elif platform_system.startswith("Win"):
+        try:
+            subprocess.run(["choco", "install", "actionlint", "-y"], check=True)
+            return True, ""
+        except (FileNotFoundError, subprocess.CalledProcessError):
+            return False, f"{error} : check Choco installation"
+    return False, error
+
+
+def install_actionlint_source(error) -> Tuple[bool, str]:
+    """Install Actionlint Binary from provided script"""
+    url = "https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash"
+    version = "1.6.17"
+    request = urllib.request.urlopen(url)
+    with open("download-actionlint.bash", "wb+") as fp:
+        fp.write(request.read())
+    try:
+        subprocess.run(["bash", "download-actionlint.bash", version], check=True)
+        return True, os.getcwd()
+    except (FileNotFoundError, subprocess.CalledProcessError):
+        return False, error
```

Comment:
❓ According to the docs for the actionlint script, it downloads the binary to the current path, or if you provide it a path in an argument. I don't see that supplied here, so how does actionlint end up into the system path? https://github.com/rhysd/actionlint/blob/v1.7.6/docs/install.md#download-script
> This script downloads actionlint (or actionlint.exe on Windows) binary to the current working directory. When you need to put the downloaded binary to some other directory, please give the directory path to the 2nd command line argument. The following example installs the latest version to /usr/bin.

Reply:
If actionlint has to be installed from source, it uses the current directory  on line 88

Reply:
Ah, I see that now. Thanks :) LGTM